src/D2L.Bmx/OktaAuthenticator.cs

```diff
@@ -59,7 +59,12 @@ bool ignoreCache
 			return new( Org: org, User: user, Client: oktaAuthenticated );
 		}
 
-		if( browserLauncher.TryGetPathToBrowser( out string? browserPath ) ) {
+		if(
+			// `TryGetPathToBrowser` does return `false` for Linux, but excluding Linux earlier here helps
+			// the compiler trim more unused code (e.g. all of PuppeteerSharp)
+			!OperatingSystem.IsLinux()
```

Comment:
I get how that works in theory, how the compiler can suss out that during build time that "hey this check with a static will always be false here, no need to include code only used in this branch" 

It is still mind blowing :P

Not relevant here but, do you know if there is any ways to guarantee that certain things are trimmed or else to throw errors etc...? AFAIK its a bit of a problem with code that relies on compiler optimizations via clever syntax, where it would silently start using the slower path after relatively innocuous looking changes.

Reply:
I don't know of a way to do that in general.

In the specific case of OS dependent code though, using [preprocessor directives](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives) is a common way to do it, e.g.
https://stackoverflow.com/questions/30153797/c-sharp-preprocessor-differentiate-between-operating-systems
The .NET runtime and core libraries themselves use this a lot.

I tried this a bit in the early days of .NET BMX, but couldn't get the dev experience I wanted.